src/msw/window.cpp

```diff
@@ -3707,57 +3707,68 @@ wxWindowMSW::MSWHandleMessage(WXLRESULT *result,
         case WM_NCCALCSIZE:
             {
                 const wxBorder border = TranslateBorder(GetBorder());
+
                 if (wxUxThemeIsActive() && border == wxBORDER_THEME)
                 {
+                    RECT newrcClient; // new client rect
+
                     // first ask the widget to calculate the border size
                     rc.result = MSWDefWindowProc(message, wParam, lParam);
                     processed = true;
 
-                    // now alter the client size making room for drawing a
-                    // themed border
-                    RECT *rect;
-                    NCCALCSIZE_PARAMS *csparam = NULL;
-                    if ( wParam )
-                    {
-                        csparam = (NCCALCSIZE_PARAMS *)lParam;
-                        rect = &csparam->rgrc[0];
-                    }
-                    else
+                    if (wParam)
                     {
-                        rect = (RECT *)lParam;
+                        LPNCCALCSIZE_PARAMS pnccsp = (LPNCCALCSIZE_PARAMS)lParam;
+                        // pnccsp->rgrc[0] is the computed new client rect
+                        // pnccsp->rgrc[1] is going to be the dst blit rect
+                        // pnccsp->rgrc[2] is going to be the src blit rect
+
+						//::CopyRect(&newrcClient, &pnccsp->rgrc[0]);
+                        newrcClient = pnccsp->rgrc[0];
                     }
+                    else // wParam == 0: Windows wants us to map a single rect w->c
+						//::CopyRect(&newrcClient, (LPRECT)lParam);
+                        newrcClient = *(LPRECT)lParam;
 
                     wxUxThemeHandle hTheme((const wxWindow *)this, L"EDIT");
-                    RECT rcClient = { 0, 0, 0, 0 };
                     wxClientDC dc((wxWindow *)this);
-                    wxMSWDCImpl *impl = (wxMSWDCImpl*) dc.GetImpl();
+                    wxMSWDCImpl *impl = (wxMSWDCImpl*)dc.GetImpl();
 
-                    if ( ::GetThemeBackgroundContentRect
-                                (
-                                 hTheme,
-                                 GetHdcOf(*impl),
-                                 EP_EDITTEXT,
-                                 ETS_NORMAL,
-                                 rect,
-                                 &rcClient) == S_OK )
+                    RECT rcContent = newrcClient;
+                    //::CopyRect(&rcContent, &newrcClient);
+
+                    // Get the state ID
+                    int nState;
+
+                    if (!IsEnabled())
+                        nState = ETS_DISABLED;
+                    // Should we check this?
+                    //else if (::GetWindowLong(GetHwnd(), GWL_STYLE) & ES_READONLY)
+                    //    nState = ETS_READONLY;
+                    else
+                        nState = ETS_NORMAL;
+
+                    ::GetThemeBackgroundContentRect(hTheme, GetHdcOf(*impl), EP_EDITTEXT,
```

Comment:
YES, it is intentional.
EXPLANATION:
In the original code `RECT rcClient = { 0, 0, 0, 0 };` was initialized with **zero**, and if `GetThemeBackgroundContentRect` fails, we'll get **zero** `rcClient`, and the following code won't be executed:
`InflateRect(&rcClient, -1, -1);

if (wParam)
    csparam->rgrc[0] = rcClient;
else
    *((RECT*)lParam) = rcClient;
`
and this is the main cause of the issue.

In the fixed code, we don't care about the return value of `GetThemeBackgroundContentRect`, because if it fails, it wont alter `rcContent` at all, and we'll still have the previous value of it that was initially copied from `newrcClient`:
`RECT rcContent = newrcClient;`.